src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -17,22 +17,23 @@
      * @param type - any class of Machine sub class. For example: Track.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Machine> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
That will make more sense to use T here, that will make code more flexible, because we assign T while declaration of class that will implement this interface